codec/testdata/proto.proto

```diff
@@ -48,3 +48,47 @@ message SayHelloRequest {
 message SayHelloResponse {
   string greeting = 1;
 }
+
+message Customer1 {
+    int32 id = 1;
+    string name = 2;
+    float subscription_fee = 3;
+
+    string payment = 7;
+}
```

Comment:
We should have `Any` and `oneof` fields in `Customer1` and they should get populated with data that should get rejected. We want tests that have unknown fields in nested messages.

```suggestion
    Customer1 nested_customer = 8;
    google.protobuf.Any misc = 9;
    oneof sum {
      int64 a = 1;
      Customer1 b = 2;
    }
}
```

Reply:
I instead went a different route and explicitly added tests for flat vs nested structs. That way we can explore as many cases without too many distractions.